### PR TITLE
CR-1123500 CR-1123180 Fix reading SDR sensor data

### PR DESCRIPTION
**Pull request by rajkumar-xilinx**
Fixed board power & temperature readings.

### CODE REVIEW

**Comment by rajkumar-xilinx** [comment]
bash-4.2# xbutil examine -r mechanical -d

---
1/1 [0000:d8:00.1] : xilinx_vck5000_gen4x8_xdma_base_1
----
Mechanical
  Fans
    Fan RPM
      Critical Trigger Temp : 61 C
      Speed                 : 10434 RPM

File: src/runtime_src/core/common/sensor.cpp [hunk 1]
```diff
@@ -262,10 +261,18 @@ read_data_driven_mechanical(std::vector<xq::sdm_sensor_info::data_type> output)
     pt.put("location_id", tmp.label);
     pt.put("description", tmp.label);
     pt.put("speed_rpm", tmp.input);
-    pt.put("critical_trigger_temp_C", "N/A");
     pt.put("is_present", "true");
-    fan_array.push_back({"", pt});
   }
+  // iterate over output data, store it into property_tree
+  for(const auto& tmp : temp)
```
Comment (stsoe) [inline src/runtime_src/core/common/sensor.cpp:267]:
```suggestion
  for (const auto& tmp : temp)
```

File: src/runtime_src/core/common/sensor.cpp [hunk 2]
```diff
@@ -262,10 +261,18 @@ read_data_driven_mechanical(std::vector<xq::sdm_sensor_info::data_type> output)
...
+  {
+    if (!strcmp("Vccint Temp", (tmp.label).c_str()))
+    {
```
Comment (stsoe) [inline src/runtime_src/core/common/sensor.cpp:270]:
```suggestion
    if (tmp.label == "Vccint Temp")  {
```

File: src/runtime_src/core/common/sensor.cpp [hunk 3]
```diff
@@ -234,13 +234,11 @@ read_data_driven_thermals(const std::vector<xq::sdm_sensor_info::data_type>& out
   // iterate over temperature data, store to ptree by converting to Celcius
   for(const auto& tmp : output)
   {
```
Comment (stsoe) [inline src/runtime_src/core/common/sensor.cpp:236]:
```suggestion
  for (const auto& tmp : output) {
```

**Review by stsoe — CHANGES_REQUESTED** [review]
Minor style requests

**Review by chvamshi-xilinx — APPROVED** [review]

**Comment by rajkumar-xilinx** [comment]
retest this please

File: src/runtime_src/core/common/sensor.cpp [hunk 4]
```diff
@@ -232,15 +233,13 @@ read_data_driven_thermals(const std::vector<xq::sdm_sensor_info::data_type>& out
...
-    uint64_t temp_C;
     ptree_type pt;
     pt.put("location_id", tmp.label);
     pt.put("description", tmp.label);
-    temp_C = std::stoull(xrt_core::utils::format_base10_shiftdown3(tmp.input));
-    pt.put("temp_C", temp_C);
-    pt.put("highest", xrt_core::utils::format_base10_shiftdown3(tmp.highest));
+    pt.put("temp_C", tmp.input);
+    pt.put("highest", tmp.highest);
```
Comment (rozumx) [inline src/runtime_src/core/common/sensor.cpp:242]: Hat does this value represent? "highest" what?

File: src/runtime_src/core/common/sensor.cpp [hunk 5]
```diff
@@ -250,22 +249,31 @@ read_data_driven_thermals(const std::vector<xq::sdm_sensor_info::data_type>& out
...
   {
     pt.put("location_id", tmp.label);
     pt.put("description", tmp.label);
     pt.put("speed_rpm", tmp.input);
-    pt.put("critical_trigger_temp_C", "N/A");
     pt.put("is_present", "true");
-    fan_array.push_back({"", pt});
   }
+  // iterate over output data, store it into property_tree
+  for (const auto& tmp : temp)
```
Comment (rozumx) [inline src/runtime_src/core/common/sensor.cpp:268]: Earlier, Soren had recommend a code change, which you did.  
Question is: Why didn't you apply it to ALL the code you changed?

File: src/runtime_src/core/common/sensor.cpp [hunk 6]
```diff
@@ -250,22 +249,31 @@ read_data_driven_thermals(const std::vector<xq::sdm_sensor_info::data_type>& out
...
+  {
+    if (boost::iequals(tmp.label, "Vccint Temp"))
```
Comment (rozumx) [inline src/runtime_src/core/common/sensor.cpp:270]: This logic is confusing, what relationship does "Vccint Temp" have with "critical_trigger_temp_C".

File: src/runtime_src/core/common/sensor.cpp [hunk 7]
```diff
@@ -499,11 +507,13 @@ read_mechanical(const xrt_core::device * device)
   ptree_type fan_array;
   bool is_data_driven = true;
   std::vector <xq::sdm_sensor_info::data_type> output;
+  std::vector <xq::sdm_sensor_info::data_type> temp_output;
 
   //Check if requested sensor data can be retrieved in data driven model.
   try {
```
Comment (rozumx) [inline src/runtime_src/core/common/sensor.cpp:513]: This logic needs to be refactored.  Not clean and relies on side effects to determine if the data `is_data_driven`.

Nutshell: There should be a separate method that is called, something like `is_sensor_vmc_data() `and called by this method.

**Review by rozumx — CHANGES_REQUESTED** [review]
Like with your other review, please schedule meeting for us to discuss these changes.

File: src/runtime_src/core/common/sensor.cpp [hunk 8]
```diff
@@ -250,22 +249,31 @@ read_data_driven_thermals(const std::vector<xq::sdm_sensor_info::data_type>& out
...
     pt.put("speed_rpm", tmp.input);
-    pt.put("critical_trigger_temp_C", "N/A");
     pt.put("is_present", "true");
-    fan_array.push_back({"", pt});
   }
+  // iterate over output data, store it into property_tree
+  for (const auto& tmp : temp)
+  {
+    if (boost::iequals(tmp.label, "Vccint Temp"))
+    {
```
Comment (rajkumar-xilinx) [inline src/runtime_src/core/common/sensor.cpp:271]: add a comment

File: src/runtime_src/core/common/sensor.cpp [hunk 9]
```diff
@@ -250,22 +249,31 @@ read_data_driven_thermals(const std::vector<xq::sdm_sensor_info::data_type>& out
...
+      pt.put("critical_trigger_temp_C", tmp.input);
+      break;
+    }
+  }
```
Comment (rajkumar-xilinx) [inline src/runtime_src/core/common/sensor.cpp:275]: fan_temp = "N/A"
     fan_temp = tmp.input
pt.put("critical_trigger_temp_C",  fan_temp);

File: src/runtime_src/core/common/sensor.cpp [hunk 10]
```diff
@@ -516,7 +526,7 @@ read_mechanical(const xrt_core::device * device)
   }
 
   if (is_data_driven)
```
Comment (rajkumar-xilinx) [inline src/runtime_src/core/common/sensor.cpp:528]: add a new method

File: src/runtime_src/core/common/sensor.cpp (hunk 6 above)
Comment (rajkumar-xilinx) [reply src/runtime_src/core/common/sensor.cpp]: We discussed with firmware team yesterday, and they suggested not to use Vccint Temp for fan's temperature. And also confirmed this data is not available for versal.

File: src/runtime_src/core/common/sensor.cpp (hunk 7 above)
Comment (rajkumar-xilinx) [reply src/runtime_src/core/common/sensor.cpp]: Removed completely as we not longer need reading thermal sensors as part of mechanical report.

**Comment by rajkumar-xilinx** [comment]
Hi @rozumx 
We discussed with firmware team as part of CRs review meeting yesterday, got to know that fan temperature is not available on versal, so removed "Vccint Temp" logic from PR. So, "Critical Trigger Temp" will be N/A only.

Updated PR with all the suggestions. Please review it again. Thank you.

**Review by rozumx — APPROVED** [review]

**Comment by rajkumar-xilinx** [comment]
Hi @stsoe could you please review this change, I have taken care of your suggestions. Thank you.

**Review by stsoe — APPROVED** [review]

**Pull request merged**